src/metric_model_reporter.cc

```diff
@@ -183,6 +184,14 @@ MetricModelReporter::~MetricModelReporter()
   }
 }
 
+std::map<FailureReason, std::string> failure_reasons_map = {
+    {FailureReason::REJECTED, "REJECTED"},
+    {FailureReason::CANCELED, "CANCELED"},
+    {FailureReason::BACKEND, "BACKEND"},
+    {FailureReason::TIMEOUT, "TIMEOUT"},
+    {FailureReason::OTHER, "OTHER"}
+};
```

Comment:
We should probably make this `static const` somewhere, not sure where yet. Let me know if you have any ideas. I'm thinking either `constants.h` or maybe a member variable of the MetricModelReporter